release/v1alpha1/types.go

```diff
@@ -0,0 +1,171 @@
+	// determine if the Source imagestream has any new images
+	Hash string `json:"hash,omitempty"`
+}
+
+// ReleasePayloadStatus the status of all the promotion test jobs
+type ReleasePayloadStatus struct {
+	// Conditions communicates the state of the ReleasePayload.
+	//
+	// Supported conditions include Pending, Ready, Failed, Accepted, and Rejected.
+	//
+	// If Pending is true the release tag is waiting for an updated payload to be created and pushed
+	// Transitions to Failed or Ready
+	//
+	// If Failed is true a ReleasePayload image cannot be created for the given set of image mirrors
+	// This condition is terminal
+	//
+	// if Ready is true a ReleasePayload has a valid update payload image created and pushed to the
+	// release image stream.  Verification jobs should begin and will update the status as they
+	// complete
+	// Transitions to Accepted or Rejected
+	//
+	// if Accepted is true the ReleasePayload has passed its verification criteria and can safely
+	// be promoted to an external location
+	// This condition is terminal
+	//
+	// if Rejected is true the ReleasePayload has failed one or more of its verification criteria
+	// The release-controller will take no more action in this phase, but a human may set the
+	// phase back to Ready to retry and the controller will attempt verification again.
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// BlockingJobResults stores the results of all blocking jobs
+	BlockingJobResults []JobStatus `json:"blockingJobResults,omitempty"`
+
+	// InformingJobResults stores the results of all informing jobs
+	InformingJobResults []JobStatus `json:"informingJobResults,omitempty"`
+
+	// AnalysisJobResults stores the results of all analysis jobs
+	AnalysisJobResults []JobStatus `json:"analysisJobResults,omitempty"`
+}
+
+// JobState the aggregate state of the job
+// Supported values include Pending, Failed, and Success.
+type JobState string
+
+const (
+	// JobStatePending not all job runs have completed
+	// Transitions to Failed or Success
+	JobStatePending JobState = "Pending"
+
+	// JobStateFailed failed job aggregation
+	JobStateFailed JobState = "Failed"
+
+	// JobStateSuccess successful job aggregation
+	JobStateSuccess JobState = "Success"
+)
+
+// JobStatus encapsulates the name of the job, all the results of the jobs, and an aggregated
+// result of all the jobs
+type JobStatus struct {
+	// jobName is the name of the job
+	JobName string `json:"name,omitempty"`
+
+	// MaxRetries maximum times to retry a job
+	MaxRetries int `json:"maxRetries,omitempty"`
+
+	// AggregateState is the overall success/failure of all the executed jobs
+	AggregateState JobState `json:"state,omitempty"`
+
+	// JobRunResults contains the links for individual jobs
+	JobRunResults []JobRunResult `json:"results,omitempty"`
+}
+
+// JobRunState the status of a job
+type JobRunState string
+
+const (
+	// JobRunStateTriggered job has been created but not scheduled
```

Comment:
give me a full resource.group please. job.batch?

Reply:
Some of these are Jobs, some are ProwJobs - might be good to differentiate :)